pkg/runner/step_runner.go

```diff
@@ -16,15 +16,12 @@ import (
 	"github.com/linuxboot/contest/pkg/xcontext"
 )
 
-type AddTargetToStep func(ctx xcontext.Context, tgt *target.Target) error
-
-type StepRunnerEvent struct {
-	// Target if set represents the target for which event was generated
-	Target *target.Target
-	// Err if Target is not nil refers to this Target result otherwise is execution error
-	Err error
+type ResultNotifier interface {
+	ResultCh() <-chan error
```

Comment:
nit;
```suggestion
	NotifyCh() <-chan error
```
and rename ResultNotifier to ChanNotifier or something; there's nothing in this that actually says "results". it's just a chan error